akka/src/main/scala/cakesolutions/kafka/akka/KafkaProducerActor.scala

```diff
@@ -142,11 +234,25 @@ private class KafkaProducerActor[K, V](
     }
   }
 
-  private def logSendError(err: Throwable): Unit =
-    log.error(err, "Failed to send message to Kafka!")
+  private def logSendError(err: Throwable): Unit = log.error(err, "Failed to send message to Kafka!")
 
   private def sendMany(records: Records) = Future.sequence(records.map(send))
 
+  private def sendManyTransactional(records: Records): Future[Iterable[RecordMetadata]] =
+    Future
+      .fromTry(Try(producer.beginTransaction()))
+      .flatMap(_ =>
+        (for {
+          results <- sendMany(records)
+          _ = Future.fromTry(Try(producer.commitTransaction()))
```

Comment:
This a bit confusing, so it is fine to commit the transaction but lose the error? If not I'd suggest
```
_ <- Future.fromTry(Try(producer.commitTransaction()))
```

Reply:
fixed